client/worker.go

```diff
@@ -54,56 +61,45 @@ func newWorker(client *http.Client, endpoint string, accessToken string, disable
 }
 
 func (w *worker) export(ctx context.Context, batches []*jaegerpb.Batch, accessToken string) *ErrSend {
-	ctx, span := trace.StartSpan(ctx, "export")
+	ctx, span := w.tracer.Start(ctx, "export")
 	defer span.End()
 
 	var spansCount int
 	for _, batch := range batches {
 		spansCount += len(batch.Spans)
 	}
 
-	span.AddAttributes(trace.Int64Attribute("spans", int64(spansCount)))
-	span.AddAttributes(trace.Int64Attribute("batches", int64(len(batches))))
+	span.SetAttributes(attribute.Int64("spans", int64(spansCount)))
+	span.SetAttributes(attribute.Int64("batches", int64(len(batches))))
+
 	if spansCount == 0 {
 		return nil
 	}
 
-	sr, err := w.prepare(ctx, batches, spansCount)
+	sr, err := w.prepare(batches, spansCount)
 	if err != nil {
+		span.SetStatus(codes.Error, err.Error())
 		recordEncodingFailure(ctx, sr)
-		span.SetStatus(trace.Status{
-			Code: trace.StatusCodeInvalidArgument,
-		})
-		return &ErrSend{Err: err}
+		return &ErrSend{Err: err, Permanent: true}
```

Comment:
Adding `Permanent` seems like unrelated fix. Can you at least mention it to the PR/commit comment.

Reply:
https://github.com/signalfx/sapm-proto/pull/56 for this